### PR TITLE
Parallel execution of server based tests

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Allow server to be started on random http/https port.
Update tests to be able to resolve port on which server is running.
Register http and https connector ports in port register component.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@lutovich can you please take a look? thx!
cc @lassewesth

**Comment by lassewesth** [comment]
@MishaDemianenko Wow, thanks massive, well done, looks good!

File: community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java [hunk 1]
```diff
@@ -375,6 +351,20 @@ protected void startJetty() throws Exception
         }
     }
 
+    @Override
+    public InetSocketAddress getLocalHttpAddress()
+    {
+        return new InetSocketAddress( serverConnector.getHost(), serverConnector.getLocalPort() );
```
Comment (lutovich) [inline community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java]: Maybe extract a method like `InetSocketAddress toSocketAddress(ServerConnector)` to use here and in `#getLocalHttpsAddress()`?

**Review by lutovich — APPROVED** [review]
@MishaDemianenko looks great!

File: community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java (hunk 1 above)
Comment (MishaDemianenko) [reply community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java]: @lutovich absolutely, thx! updated!

**Pull request merged**